lib/bolt/pal.rb

```diff
@@ -107,6 +107,18 @@ def in_bolt_compiler
       r = Puppet::Pal.in_tmp_environment('bolt', modulepath: @modulepath, facts: {}) do |pal|
         pal.with_script_compiler do |compiler|
           alias_types(compiler)
+          
+          internal_compiler = compiler.instance_variable_get(:@internal_compiler)
+          loaders = internal_compiler.loaders
+          # # Load built in types
+          loaders.static_loader.runtime_3_init
+          # Load types from modulepath
+          puppet_system_loader = internal_compiler.loaders.puppet_system_loader
```

Comment:
This bit here attempts to set up the runtime3 loader, but it does not actually end up resolving the types.